modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/sql/ClientSqlCursorNextResultRequest.java

```diff
@@ -60,5 +61,26 @@ public static CompletableFuture<ResponseWriter> process(
                         ).thenCompose(asyncResultSet ->
                                 ClientSqlCommon.writeResultSetAsync(resources, asyncResultSet, metrics, pageSize, false, false, true)
                         ).thenApply(rsWriter -> rsWriter), operationExecutor);
+
+        f.whenCompleteAsync((r, t) -> {
+            if (t != null) {
+                cursorWithPageSize.cursorFuture().thenAccept(cur -> closeRemainingCursors(cur, false, operationExecutor));
```

Comment:
Seems, here `cursorWithPageSize.cursorFuture()` is already completed,
Because f can't be completed earlier.